annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,158 @@
+from unittest.mock import Mock, call, patch
+
+from annotation.models import File
+import pytest
+from sqlalchemy.orm import Session
+
+from annotation.errors import FieldConstraintError
+from annotation.jobs.services import ValidationSchema
+from annotation.tasks.services import validate_files_info, validate_task_info, validate_users_info
+
+
+TEST_PARAMS_INVALID_FILES = [
+    (
+        {
+            "file_id": 1,
+            "job_id": 2,
+            "pages": [1, 2, 4],
+        },
+        Mock(spec=File, pages_number=3),
+        "pages ({4}) do not belong to file"
+    ),
+    (
+        {
+            "file_id": 1,
+            "job_id": 2,
+            "pages": [1, 2, 3],
+        },
+        None,
+        "file with id 1 is not assigned for job 2"
+    ),
+]
+
+
+@pytest.fixture(scope="function")
+def db_session():
+    return Mock(spec=Session)
+
+@pytest.fixture(scope="function")
+def task_info():
+    return {
+        "is_validation": True,
+        "user_id": 1,
+        "job_id": 2,
+    }
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    (
+        (ValidationSchema.validation_only, True),
+        ("other_type", False),
+        ("other_type", True),
+    ),
+)
+def test_validate_task_info_positive_case(
+    db_session: Session, task_info: dict, validation_type: ValidationSchema, is_validation: bool
+):
+    task_info["is_validation"] = is_validation
+
+    with patch(
+        "annotation.tasks.services.validate_users_info"
+    ) as mock_validate_users_info, patch(
+        "annotation.tasks.services.validate_files_info"
+    ) as mock_validate_files_info:
+        validate_task_info(db_session, task_info, validation_type)
+        mock_validate_users_info.assert_called_once_with(
+            db_session, task_info, validation_type
+        )
+        mock_validate_files_info.assert_called_once_with(
+            db_session, task_info
+        )
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    (
+        (ValidationSchema.validation_only, False),
+    ),
+)
+def test_validate_task_info_negative_case(
+    db_session: Session,task_info: dict, validation_type: ValidationSchema, is_validation: bool
```

Comment:
```suggestion
    db_session: Session, task_info: dict, validation_type: ValidationSchema, is_validation: bool
```

Reply:
and specify dict nested types

Reply:
Instead of `negative_case` specify explanatory name of what you do here/when such cases appear

Reply:
Ok :)